js/editor.js

```diff
@@ -1 +1,33 @@
-// Редактирование изображения editor.js
+import { onImgUploadOverlayCancelButtonClick, onImgUploadOverlayEscKeydown } from './render-thumbnails.js';
+
+// Форма загрузки изображения на сайт
+const uploadForm = document.querySelector('.img-upload__form');
+// Поле загрузки изображения
+const uploadFileInput = uploadForm.querySelector('#upload-file');
+// Попап (форма) редактирования загружаемого изображения
+const imgUploadOverlay = uploadForm.querySelector('.img-upload__overlay');
+// Кнопка закрытия оверлея
+const imgUploadOverlayCancellButton = uploadForm.querySelector('#upload-cancel');
+
+// Функция закрытия оверлея редактирования загружаемого изображения
+const closeUploadOverlay = () => {
+
+  imgUploadOverlay.classList.add('hidden');
+  document.body.classList.remove('modal-open');
+
+  uploadForm.reset();
+  document.removeEventListener('keydown', onImgUploadOverlayEscKeydown);
+  imgUploadOverlayCancellButton.removeEventListener('click', onImgUploadOverlayCancelButtonClick);
+};
+
+// Событие изменения поля загрузки изображения
+uploadFileInput.addEventListener('change', () => {
+  imgUploadOverlay.classList.remove('hidden');
+  document.body.classList.add('modal-open');
+
+
+  document.addEventListener('keydown', onImgUploadOverlayEscKeydown);
+  imgUploadOverlayCancellButton.addEventListener('click', onImgUploadOverlayCancelButtonClick);
+});
+
+export {closeUploadOverlay};
```

Comment:
😟

<details>
<summary>editor.js</summary>

https://github.com/htmlacademy-javascript/2049181-kekstagram-26/blob/b387a3217313c67cc6148778d1d051642dec9d9b/js/editor.js#L1
https://github.com/htmlacademy-javascript/2049181-kekstagram-26/blob/b387a3217313c67cc6148778d1d051642dec9d9b/js/editor.js#L33
</details>

<details>
<summary>render-thumbnails.js</summary>

https://github.com/htmlacademy-javascript/2049181-kekstagram-26/blob/b387a3217313c67cc6148778d1d051642dec9d9b/js/render-thumbnails.js#L4
https://github.com/htmlacademy-javascript/2049181-kekstagram-26/blob/b387a3217313c67cc6148778d1d051642dec9d9b/js/render-thumbnails.js#L55
</details>

Модули `***.js` и `***.js` напрямую вызывают методы друг друга и таким образом непосредственно зависят друг от друга; взаимозависимые модули не являются полностью отдельными модулями, будучи по сути частями одного большего модуля, что создает опасную ситуацию [циклической зависимости](https://en.wikipedia.org/wiki/Circular_dependency):

> In software design, **circular dependencies** between larger software modules are considered an [anti-pattern](https://ru.wikipedia.org/wiki/%D0%90%D0%BD%D1%82%D0%B8%D0%BF%D0%B0%D1%82%D1%82%D0%B5%D1%80%D0%BD) because of their negative effects

<details>
<summary>Еще мнение</summary>

Если нужно еще мнение, то на днях с другим студентом обсуждали [вебинар по организации бизнес-логики](https://www.youtube.com/watch?v=nr99fgGdQoQ#t=30m2s), с мыслями в котором в целом солидарен:

> Модульность — это отсутствие циклических зависимостей

> Как только две штуки друг про друга знают, то модульность отсутствует
</details>

<details>
<summary>Что еще важно понять и помнить</summary>

Важно понимать, что [модульная система](https://ru.wikipedia.org/wiki/%D0%9C%D0%BE%D0%B4%D1%83%D0%BB%D1%8C%D0%BD%D0%BE%D0%B5_%D0%BF%D1%80%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5#%D0%9C%D0%BE%D0%B4%D1%83%D0%BB%D1%8C%D0%BD%D0%B0%D1%8F_%D1%81%D0%B8%D1%81%D1%82%D0%B5%D0%BC%D0%B0_%D0%BC%D0%BE%D0%B4%D1%83%D0%BB%D0%B5%D0%B9) сама по себе не создаёт _модульности_:

> Модульное программирование никак не связано с деталями конкретного языка (и даже в случае отсутствия явной поддержки со стороны языка может применяться при достаточной дисциплине со стороны программистов)

Так модули ES6 только _технически_ упрощают реализацию _модульности_. Даже упрощая сборщикам применение техники [tree shaking](https://en.wikipedia.org/wiki/Tree_shaking) (= включение только живого кода), по факту, её эффективность также зависит от умения организовать с помощью них _модульность_.
</details>

Reply:
> На данном этапе следует ли мне убрать эти зависимости, а проблему появления eslint[no-use-before-define] решить декларативным объявлением функции?

Да, сделать небольшой шаг назад, вернуть все обработчики в модуль непосредственного использования, а проблему `eslint[no-use-before-define]` решить декларативным объявлением функции.